go/service/chat_local.go

```diff
@@ -251,6 +286,68 @@ func (h *chatLocalHandler) ResolveConversationLocal(ctx context.Context, arg cha
 	}, nil
 }
 
+func (h *chatLocalHandler) resolveConversationsPipeline(ctx context.Context, convs []chat1.Conversation) ([]chat1.ConversationLocal, error) {
+	// Fetch conversation local information in parallel
+	ctx, _ = getUserInfoMapper(ctx, h.G())
+	type jobRes struct {
+		conv  chat1.ConversationLocal
+		index int
+	}
+	type job struct {
+		conv  chat1.Conversation
+		index int
+	}
+	eg, ctx := errgroup.WithContext(ctx)
+	convCh := make(chan job)
```

Comment:
This `errgroup` thing is so nice! Although why having the two channels? Can we do something like this:

```go
for i, conv := range convs {
  eg.Go(func() err {
    select {
      case <-ctx.Done():
        return ctx.Error()
      default:
    }
    info, maxMessages, err := h.getConversationInfo(ctx, conv.conv)
    if err != nil {
      return err
    }
    res = append(res, chat1.ConversationLocal{
      Info: &info, 
      Messages: maxMessages,
      ReadUpTo: conv.conv.ReaderInfo.ReadMsgid,
    })
  })
}
if err = eg.Wait() {
  return nil, err
}
return res, nil